### PR TITLE
Carry call type in Mesos client for debugging.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This introduces the call type as a context to the Mesos client flow.
This should ease debugging.

### CODE REVIEW

**Comment by jeschkies** [comment]
Pin @timcharper

**Comment by jeschkies** [comment]
Ping @timcharper

**Comment by jeschkies** [comment]
Ping @timcharper

File: mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala [hunk 1]
```diff
@@ -405,26 +405,27 @@ class MesosClientImpl(
...
-      }
+  private val responseHandler: Sink[(HttpResponse, Call.Type), Future[Done]] =
+    Sink.foreach[(HttpResponse, Call.Type)] {
+      case (response, callType) =>
+        if (response.status.isFailure()) {
+          logger.info(s"A request to Mesos failed with response: ${response.status}")
+          response.discardEntityBytes()
+          throw new IllegalStateException(s"Failed to send $callType call to Mesos.")
+        } else {
+          logger.info(s"Mesos $callType call response: $response")
```
Comment (timcharper) [inline mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala]: this should probably stay debug level, right?

**Review by timcharper — APPROVED** [review]

File: mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala (hunk 1 above)
Comment (jeschkies) [inline mesos-client/src/main/scala/com/mesosphere/mesos/client/MesosClient.scala]:
```suggestion
          logger.debug(s"Mesos $callType call response: $response")
```

**Pull request merged**